components/browser/menu/src/test/java/mozilla/components/browser/menu/WebExtensionBrowserMenuTest.kt

```diff
@@ -434,7 +431,6 @@ class WebExtensionBrowserMenuTest {
         val menu: WebExtensionBrowserMenu = mock()
 
         menuItem.bind(menu, view)
-        testDispatcher.advanceUntilIdle()
```

Comment:
Where these unneeded? I see that below in `WebExtensionBrowserMenuItemTest` you kept the calls.

Reply:
Nice catch!
Cannot say I remember this exactly but looking at it now the interesting part is the [`bind` method](https://github.com/mozilla-mobile/android-components/blob/6c0fe91ab83614265f533b819bf406fc046b2227/components/browser/menu/src/main/java/mozilla/components/browser/menu/item/WebExtensionBrowserMenuItem.kt#L52) which gets to execute `setupIcon` - launching  a new coroutine to set an icon to the item:
```
MainScope().launch {
    loadIcon(view.context, imageView.measuredHeight)?.let {
        iconTintColorResource?.let { tint -> imageView.setTintResource(tint) }
        imageView.setImageDrawable(it)
    }
}
```

The two tests from `WebExtensionBrowserMenuTest` from where I removed `advanceUntilIdle` check only how many items are shown and what happens when they are clicked without the need to actually complete the `bind` method, or other similar setups since their result is not checked. 
The tests from `WebExtensionBrowserMenuItemTest` do granular checks of the results from executing `bind(..)` really only just two tests probably needing the `advanceUntilIdle` calls - where the tint and the image drawable is checked.
With this in mind if I had to refactor again I'd still probably keep all the previous `advanceUntilIdle` calls here just to move swiftly with the changes, avoid going too deep in the functionality and really avoid any regression risk.

Reply:
@Mugurell OK, I went over all the `advanceUntilIdle` calls you removed, and they do indeed all seem unnecessary! Thanks.